src/CloudModelGenerator/Program.cs

```diff
@@ -32,6 +36,85 @@ static int Main(string[] args)
             return Execute(syntax);
         }
 
+        internal static string[] CorrectArguments(string[] args)
```

Comment:
This method is really hard to read.
Could you split it into the methods to make it more readable? It would probably help to for cover them by tests separately.

Do we want to have this method in this file, or separate it into another file to not have it next t the main method?